rmgpy/data/kinetics/kineticsTest.py

```diff
@@ -1091,3 +1091,97 @@ def test_add_atom_labels_for_reaction_3(self):
         self.assertIn('*2',found_labels)
         self.assertIn('*3',found_labels)
 
+    def test_species_preserved_after_generate_reactions(self):
+        """Test that Species objects do not retain changes after generating reactions"""
+        r1 = Species(index=1, label='ethyl', SMILES='C[CH2]')
+        r1_copy = r1.copy(deep=True)  # These copies record the state of the original attributes
+        expected_product_1 = Species(SMILES='CC')
+        expected_product_2 = Species(SMILES='C=C')
+
+        reaction_list = self.database.kinetics.generate_reactions_from_families(
+            [r1, r1], only_families=['Disproportionation'], resonance=True
+        )
+
+        # First confirm that we get the expected reaction
+        self.assertEqual(len(reaction_list), 1)
+        reaction = reaction_list[0]
+        case_1 = reaction.products[0].isIsomorphic(expected_product_1) and reaction.products[1].isIsomorphic(expected_product_2)
+        case_2 = reaction.products[0].isIsomorphic(expected_product_2) and reaction.products[1].isIsomorphic(expected_product_1)
+        # Only one case should be true
+        self.assertTrue(case_1 or case_2)
+
+        r1_out, r2_out = reaction.reactants
+
+        # The species in the output reaction should be new objects because we made them from Molecule objects
+        self.assertIsNot(r1, r1_out)
+        self.assertIsNot(r1, r2_out)
+
+        # The molecule objects should be the same references for the first output species
+        self.assertIsNot(r1.molecule[0], r1_out.molecule[0])
```

Comment:
Why is one of them the molecule objects the same reference and the other isn't?  Why can't all identical molecule objects be the same reference?

Reply:
If they were the same object then you couldn't assign different labels to the atoms.